migration_original/ODS1Stage/tables/Show/SOLRLineOfService/spu_translated_SOLRLineOfService.sql

```diff
@@ -0,0 +1,138 @@
+-- Show_spuSOLRLineOfServiceGenerateFromMid
+CREATE OR REPLACE PROCEDURE SHOW.SP_LOAD_SOLRLINEOFSERVICE() 
+    RETURNS STRING
+    LANGUAGE SQL
+    EXECUTE AS CALLER
+    AS  
+
+DECLARE 
+
+---------------------------------------------------------
+--------------- 0. Table dependencies -------------------
+---------------------------------------------------------
+    
+-- Show.SOLRLineOfService depends on: 
+--- Mid.LineOfService
+--- Show.SOLRLineOfServiceDelta  
+
+---------------------------------------------------------
+--------------- 1. Declaring variables ------------------
+---------------------------------------------------------
+
+    select_statement STRING; -- CTE statement
+    update_statement STRING;
+    insert_statement STRING;
+    merge_statement STRING; -- Insert statement to final table
+    status STRING; -- Status monitoring
+   
+---------------------------------------------------------
+--------------- 2.Conditionals if any -------------------
+---------------------------------------------------------   
+   
+BEGIN
+    -- no conditionals
+
+
+---------------------------------------------------------
+----------------- 3. SQL statements ---------------------
+---------------------------------------------------------     
+
+-- Select Statement
+select_statement :=
+'SELECT
+        midLine.LineOfServiceID,
+        midLine.LineOfServiceCode,
+        midLine.LineOfServiceTypeCode,
+        midLine.LineOfServiceDescription,
+        midLine.LegacyKey,
+        midLine.LegacyKeyName,
+        CURRENT_TIMESTAMP AS UpdatedDate,
+        CURRENT_USER AS UpdatedSource
+    FROM
+        Mid.LineOfService midLine
+    WHERE
+        midLine.LineOfServiceID IN (
```

Comment:
Replace the subquery with a CTE and the where with a JOIN

Reply:
WITH cte_id AS (
            SELECT
                DISTINCT LineOfServiceID
            FROM
                Show.SOLRLineOfServiceDelta
            WHERE
                StartDeltaProcessDate IS NULL
                AND EndDeltaProcessDate IS NULL
                AND SolrDeltaTypeCode = 1 --INSERT/UPDATEs
                AND MidDeltaProcessComplete = 1
)
SELECT
        midLine.LineOfServiceID,
        midLine.LineOfServiceCode,
        midLine.LineOfServiceTypeCode,
        midLine.LineOfServiceDescription,
        midLine.LegacyKey,
        midLine.LegacyKeyName,
        CURRENT_TIMESTAMP AS UpdatedDate,
        CURRENT_USER AS UpdatedSource
    FROM
        Mid.LineOfService midLine
    JOIN
        cte_id ON midLine.LineOfServiceID = cte_id.LineOfServiceID